### PR TITLE
Add steve API and bundle

### PR DESCRIPTION
**Pull request by CraigMChen**
#### What type of this PR

/kind feature

#### What this PR does / why we need it:

1. Integrate steve server in cmp.
2. Add bundle for steve server.
3. Support audit, authentication and cache.

#### Specified Reviewers:

/assign @johnlanni

### CODE REVIEW

File: apistructs/steve.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/erda-project/erda/pkg/strutil"
+)
+
+const (
+	SteveErrorType = "error"
+)
+
+// SteveCollection 用于接收steve server返回的集合类型的数据（一次返回多条数据）
```
Comment (dspo) [inline apistructs/steve.go:26]: It is best to comment in English

File: bundle/steve_test.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package bundle
+
+//import (
```
Comment (dspo) [inline bundle/steve_test.go:16]: do not submit invalid code

File: modules/cmp/endpoints/steve.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+)
+
+const (
+	// clusterTypeK8S Identify the k8s cluster type in the colony event
+	clusterTypeK8S = "k8s"
+)
+
+// SteveClusterHook starts steve server when create cluster and stop steve server when delete cluster
+func (e *Endpoints) SteveClusterHook(ctx context.Context, r *http.Request, vars map[string]string) (resp httpserver.Responser, err error) {
+	req := apistructs.ClusterEvent{}
```
Comment (dspo) [inline modules/cmp/endpoints/steve.go:37]: `var req apistructs.ClusterEvent`

File: modules/cmp/endpoints/steve.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
```
Comment (dspo) [inline modules/cmp/endpoints/steve.go:38]: the `Decode` will be panic if `r.Body == nil`

File: modules/cmp/endpoints/steve.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+		errstr := fmt.Sprintf("decode clusterhook request fail: %v", err)
+		logrus.Error(errstr)
+		return httpserver.HTTPResponse{Status: http.StatusBadRequest, Content: errstr}, nil
```
Comment (dspo) [inline modules/cmp/endpoints/steve.go:44]:
```go
err = errors.Wrap(err, "failed to decode clusterhook request")
logrus.Error(err)
return httpserver.HTTPResponse(Status: http.StatusBadRequest, Content: err.Error()), nil
```
reference: https://yuque.antfin.com/dice/zs3zid/dkr3ym#456d29ef
"The action log uses the following format: failed to do something"

File: modules/cmp/initialize.go [hunk 6]
```diff
@@ -116,7 +116,7 @@ func do() (*httpserver.Server, error) {
 	bundleOpts := []bundle.Option{
 		bundle.WithHTTPClient(
 			httpclient.New(
-				httpclient.WithTimeout(time.Second, time.Second*60),
+				httpclient.WithTimeout(30*time.Second, time.Second*60),
```
Comment (dspo) [inline modules/cmp/initialize.go:119]: To unify the format, either all are `time.Second * num`, or all are `num * time.Second`
```go
httpclient.WithTiemeout(time.Second * 30, time.Second * 60)
```

File: modules/cmp/initialize.go [hunk 7]
```diff
@@ -139,13 +140,21 @@ func do() (*httpserver.Server, error) {
...
+	}
+
 	if conf.EnableEss() {
 		initServices(ep)
 	}
 
 	server := httpserver.New(conf.ListenAddr())
 	server.RegisterEndpoint(append(ep.Routes()))
 
+	server.Router().PathPrefix("/k8s/clusters/{clusterName}").Handler(ep.SteveAggregator)
```
Comment (dspo) [inline modules/cmp/initialize.go:156]: magic value `"/k8s/clusters/{clusterName}"`

File: modules/cmp/steve/aggregator.go [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+	c, ok := a.cancels.Load(clusterName)
+	if !ok {
+		return nil
+	}
+	cancel, _ := c.(context.CancelFunc)
+	cancel()
+	a.cancels.Delete(clusterName)
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (CraigMChen) [reply modules/cmp/steve/aggregator.go:143]: 已添加补偿机制

**Review by liuhaoyang — CHANGES_REQUESTED** [review]
New apis should be defined using proto

File: modules/cmp/initialize.go [hunk 9]
```diff
@@ -135,17 +137,20 @@ func do() (*httpserver.Server, error) {
...
+	ep, err := initEndpoints(ctx, db, js, cachedJs, bdl, o)
 	if err != nil {
 		return nil, err
 	}
+
 	if conf.EnableEss() {
 		initServices(ep)
 	}
 
 	server := httpserver.New(conf.ListenAddr())
```
Comment (liuhaoyang) [inline modules/cmp/initialize.go:149]: The new module should use Erda infra instead of obsolete endpoints

**Review by liuhaoyang — DISMISSED** [review]

**Review by johnlanni — CHANGES_REQUESTED** [review]

**Review by johnlanni — APPROVED** [review]

**Pull request merged**